Katello/katello
# Fixes #33077 - Enable Tracer via remote execution

- [x] via remote execution
- [x] via customized remote execution

This adds "Enable Tracer" empty state and modal to the Traces tab.
* allows you to enable Tracer via remote execution
* allows you to enable Tracer via customized remote execution
* Adds reusable RemoteExecution Redux actions

### To test
#### Get a working REX setup
1. In Infrastructure > Smart Proxies, verify that the 'SSH' feature is listed
2. If needed, run `foreman-installer --enable-foreman-proxy-plugin-remote-execution-ssh --disable-system-checks` on the server
1. In Settings > Authentication, turn off "Require SSL for smart proxies"
2. Spin up a `centos7-katello-client` and make sure it's registered
3. On the smart proxy / server: `cat ~foreman-proxy/.ssh/id_rsa_foreman_proxy.pub`
4. Copy the output to the clipboard
5. On the client, as root, paste the contents of the file into `~/.ssh/authorized_keys`
6. Confirm that it's working: On the smart proxy / server, `ssh -i ~foreman-proxy/.ssh/id_rsa_foreman_proxy root@target.example.com`
7. Run `yum update` on the client and you'll probably have some Traces (optional)

#### Test with Tracer installed
1. Hosts > All Hosts > (your client) > Edit dropdown > New details page > Traces tab
2. Verify that the Traces table loads normally

#### Test without Tracer installed
1. Hosts > All Hosts > (your client) > Edit dropdown > New details page > Traces tab
2. Verify that the "Enable Tracer" empty state shows
3. Click the button and try "via remote execution" > Enable Tracer
4. View the task results and then return to the Traces tab: the table should load the traces normally
5. On the client, `yum -y remote katello-host-tools-tracer` (OR you could just use REX ;)
6. Refresh the page and you should see the Enable Tracer empty state again
7. This time try the "via customized remote execution" option - you should be taken to the New Job Invocations page with the hostname and package name filled in
